tests/operators/test_autograd_linop.py

```diff
@@ -0,0 +1,91 @@
+"""Test for the adjoint as backward option in the LinearOperator class."""
+
+import pytest
+import torch
+from mrpro.operators import LinearOperator
+from torch.autograd.gradcheck import GradcheckError
+
+from tests import RandomGenerator
+from tests.helper import dotproduct_adjointness_test
+
+
+class NonDifferentiableOperator(LinearOperator, adjoint_as_backward=False):
+    """Dummy operator for testing purposes."""
+
+    def forward(self, x):
+        python_float = x.item()  # breaks autograd as floats do not track gradients
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)
+        return (tensor,)
+
+    def adjoint(self, x):
+        python_float = x.item()  # breaks autograd as floats do not track gradients
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)
+        return (tensor,)
+
+
+class DifferentiableOperator(LinearOperator, adjoint_as_backward=True):
+    """Dummy operator for testing purposes. Should be differentiable due to adjoint_as_backward=True."""
+
+    def forward(self, x):
+        python_float = x.item()  # would break autograd as floats do not track gradients
+        tensor = torch.tensor([2 * python_float], dtype=x.dtype)  # breaks the autograd
```

Comment:
I guess x.item() from above breaks the autograd and not this line

Reply:
ups, forgot to remove second line. fixed